parl/remote/client.py

```diff
@@ -195,8 +195,14 @@ def _create_job_monitor(self, job_heartbeat_socket):
             try:
                 job_heartbeat_socket.send_multipart(
                     [remote_constants.HEARTBEAT_TAG])
-                _ = job_heartbeat_socket.recv_multipart()
-                time.sleep(remote_constants.HEARTBEAT_INTERVAL_S)
+                job_message = job_heartbeat_socket.recv_multipart()
+                if to_str(job_message[1]) == 'True':
```

Comment:
It's hard to maintain the code written in this way. You should define a variable representing the meaning of `job_message[1]` and then check whether it is `True`.